pre-commit-hooks/fix-local-includes.sh

```diff
@@ -16,6 +16,16 @@ else
     main_src_dirs=$(find ./src -maxdepth 1 -type d -exec basename {} \; | paste -sd '|' | sed 's/|/\\|/g')
 fi
 
+# Get a list of all header filenames in src/ and tests/ (recursively, unique)
+all_local_headers=$(find ./src ./tests -type f \( -name '*.hpp' -o -name '*.h' \) -exec basename {} \; | sort -u)
```

Comment:
Do we need 2 implementations?
I mean using both `main_src_dirs` and `all_local_headers`?